src/jaxsim/parsers/descriptions/model.py

```diff
@@ -249,7 +249,25 @@ def __eq__(self, other: ModelDescription) -> bool:
         if not isinstance(other, ModelDescription):
             return False
 
-        return hash(self) == hash(other)
+        if self.name != other.name:
+            return False
+
+        if self.fixed_base != other.fixed_base:
+            return False
+
+        if self.root != other.root:
+            return False
+
+        if self.joints != other.joints:
+            return False
+
+        if self.frames != other.frames:
+            return False
+
+        if self.root_pose != other.root_pose:
```

Comment:
We could also make this more compact
```suggestion
        if not all(
            [
                self.name == other.name,
                self.fixed_base == other.fixed_base,
                self.root == other.root,
                self.joints == other.joints,
                self.frames == other.frames,
                self.root_pose == other.root_pose,
            ]
        ):
```